<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="generator" content="rustdoc">
    <meta name="description" content="API documentation for the Rust `google_genomics1_beta2` crate.">
    <meta name="keywords" content="rust, rustlang, rust-lang, google_genomics1_beta2">

    <title>google_genomics1_beta2 - Rust</title>

    <link rel="stylesheet" type="text/css" href="../main.css">

    
    
</head>
<body class="rustdoc">
    <!--[if lte IE 8]>
    <div class="warning">
        This old browser is unsupported and will most likely display funky
        things.
    </div>
    <![endif]-->

    

    <section class="sidebar">
        
        <p class='location'></p><script>window.sidebarCurrent = {name: 'google_genomics1_beta2', ty: 'mod', relpath: '../'};</script>
    </section>

    <nav class="sub">
        <form class="search-form js-only">
            <div class="search-container">
                <input class="search-input" name="search"
                       autocomplete="off"
                       placeholder="Click or press 'S' to search, '?' for more options..."
                       type="search">
            </div>
        </form>
    </nav>

    <section id='main' class="content mod">
<h1 class='fqn'><span class='in-band'>Crate <a class='mod' href=''>google_genomics1_beta2</a><wbr></span><span class='out-of-band'><a href='stability.html'>[stability]</a> <span id='render-detail'>
            <a id="collapse-all" href="#">[-]</a>&nbsp;<a id="expand-all" href="#">[+]</a>
        </span><a id='src-0' href='../src/google_genomics1_beta2/lib.rs.html#5-213'>[src]</a></span></h1>
<div class='docblock'><p>This documentation was generated from <em>genomics</em> crate version <em>0.1.8+20150326</em>, where <em>20150326</em> is the exact revision of the <em>genomics:v1beta2</em> schema built by the <a href="http://www.makotemplates.org/">mako</a> code generator <em>v0.1.8</em>.</p>

<p>Everything else about the <em>genomics</em> <em>v1_beta2</em> API can be found at the
<a href="https://developers.google.com/genomics/v1beta2/reference">official documentation site</a>.
The original source code is <a href="https://github.com/Byron/google-apis-rs/tree/master/gen/genomics1_beta2">on github</a>.</p>

<h1 id="features" class='section-header'><a
                           href="#features">Features</a></h1>
<p>Handle the following <em>Resources</em> with ease from the central <a href="struct.Genomics.html">hub</a> ... </p>

<ul>
<li><a href="struct.AnnotationSet.html">annotation sets</a>

<ul>
<li><a href="struct.AnnotationSetCreateCall.html"><em>create</em></a>, <a href="struct.AnnotationSetDeleteCall.html"><em>delete</em></a>, <a href="struct.AnnotationSetGetCall.html"><em>get</em></a>, <a href="struct.AnnotationSetPatchCall.html"><em>patch</em></a>, <a href="struct.AnnotationSetSearchCall.html"><em>search</em></a> and <a href="struct.AnnotationSetUpdateCall.html"><em>update</em></a></li>
</ul></li>
<li><a href="struct.Annotation.html">annotations</a>

<ul>
<li><a href="struct.AnnotationBatchCreateCall.html"><em>batch create</em></a>, <a href="struct.AnnotationCreateCall.html"><em>create</em></a>, <a href="struct.AnnotationDeleteCall.html"><em>delete</em></a>, <a href="struct.AnnotationGetCall.html"><em>get</em></a>, <a href="struct.AnnotationPatchCall.html"><em>patch</em></a>, <a href="struct.AnnotationSearchCall.html"><em>search</em></a> and <a href="struct.AnnotationUpdateCall.html"><em>update</em></a></li>
</ul></li>
<li>callsets

<ul>
<li><a href="struct.CallsetCreateCall.html"><em>create</em></a>, <a href="struct.CallsetDeleteCall.html"><em>delete</em></a>, <a href="struct.CallsetGetCall.html"><em>get</em></a>, <a href="struct.CallsetPatchCall.html"><em>patch</em></a>, <a href="struct.CallsetSearchCall.html"><em>search</em></a> and <a href="struct.CallsetUpdateCall.html"><em>update</em></a></li>
</ul></li>
<li><a href="struct.Dataset.html">datasets</a>

<ul>
<li><a href="struct.DatasetCreateCall.html"><em>create</em></a>, <a href="struct.DatasetDeleteCall.html"><em>delete</em></a>, <a href="struct.DatasetGetCall.html"><em>get</em></a>, <a href="struct.DatasetListCall.html"><em>list</em></a>, <a href="struct.DatasetPatchCall.html"><em>patch</em></a>, <a href="struct.DatasetUndeleteCall.html"><em>undelete</em></a> and <a href="struct.DatasetUpdateCall.html"><em>update</em></a></li>
</ul></li>
<li>experimental

<ul>
<li><a href="struct.ExperimentalJobCreateCall.html"><em>jobs create</em></a></li>
</ul></li>
<li><a href="struct.Job.html">jobs</a>

<ul>
<li><a href="struct.JobCancelCall.html"><em>cancel</em></a>, <a href="struct.JobGetCall.html"><em>get</em></a> and <a href="struct.JobSearchCall.html"><em>search</em></a></li>
</ul></li>
<li>readgroupsets

<ul>
<li><a href="struct.ReadgroupsetAlignCall.html"><em>align</em></a>, <a href="struct.ReadgroupsetCallCall.html"><em>call</em></a>, <a href="struct.ReadgroupsetCoveragebucketListCall.html"><em>coveragebuckets list</em></a>, <a href="struct.ReadgroupsetDeleteCall.html"><em>delete</em></a>, <a href="struct.ReadgroupsetExportCall.html"><em>export</em></a>, <a href="struct.ReadgroupsetGetCall.html"><em>get</em></a>, <a href="struct.ReadgroupsetImportCall.html"><em>import</em></a>, <a href="struct.ReadgroupsetPatchCall.html"><em>patch</em></a>, <a href="struct.ReadgroupsetSearchCall.html"><em>search</em></a> and <a href="struct.ReadgroupsetUpdateCall.html"><em>update</em></a></li>
</ul></li>
<li><a href="struct.Read.html">reads</a>

<ul>
<li><a href="struct.ReadSearchCall.html"><em>search</em></a></li>
</ul></li>
<li><a href="struct.Reference.html">references</a>

<ul>
<li><a href="struct.ReferenceBaseListCall.html"><em>bases list</em></a>, <a href="struct.ReferenceGetCall.html"><em>get</em></a> and <a href="struct.ReferenceSearchCall.html"><em>search</em></a></li>
</ul></li>
<li>referencesets

<ul>
<li><a href="struct.ReferencesetGetCall.html"><em>get</em></a> and <a href="struct.ReferencesetSearchCall.html"><em>search</em></a></li>
</ul></li>
<li>streaming readstore

<ul>
<li><a href="struct.StreamingReadstoreStreamreadCall.html"><em>streamreads</em></a></li>
</ul></li>
<li><a href="struct.Variant.html">variants</a>

<ul>
<li><a href="struct.VariantCreateCall.html"><em>create</em></a>, <a href="struct.VariantDeleteCall.html"><em>delete</em></a>, <a href="struct.VariantGetCall.html"><em>get</em></a>, <a href="struct.VariantSearchCall.html"><em>search</em></a> and <a href="struct.VariantUpdateCall.html"><em>update</em></a></li>
</ul></li>
<li>variantsets

<ul>
<li><a href="struct.VariantsetDeleteCall.html"><em>delete</em></a>, <a href="struct.VariantsetExportCall.html"><em>export</em></a>, <a href="struct.VariantsetGetCall.html"><em>get</em></a>, <a href="struct.VariantsetImportVariantCall.html"><em>import variants</em></a>, <a href="struct.VariantsetMergeVariantCall.html"><em>merge variants</em></a>, <a href="struct.VariantsetPatchCall.html"><em>patch</em></a>, <a href="struct.VariantsetSearchCall.html"><em>search</em></a> and <a href="struct.VariantsetUpdateCall.html"><em>update</em></a></li>
</ul></li>
</ul>

<p>Not what you are looking for ? Find all other Google APIs in their Rust <a href="../index.html">documentation index</a>.</p>

<h1 id="structure-of-this-library" class='section-header'><a
                           href="#structure-of-this-library">Structure of this Library</a></h1>
<p>The API is structured into the following primary items:</p>

<ul>
<li><strong><a href="struct.Genomics.html">Hub</a></strong>

<ul>
<li>a central object to maintain state and allow accessing all <em>Activities</em></li>
<li>creates <a href="trait.MethodsBuilder.html"><em>Method Builders</em></a> which in turn
allow access to individual <a href="trait.CallBuilder.html"><em>Call Builders</em></a></li>
</ul></li>
<li><strong><a href="trait.Resource.html">Resources</a></strong>

<ul>
<li>primary types that you can apply <em>Activities</em> to</li>
<li>a collection of properties and <em>Parts</em></li>
<li><strong><a href="trait.Part.html">Parts</a></strong>

<ul>
<li>a collection of properties</li>
<li>never directly used in <em>Activities</em></li>
</ul></li>
</ul></li>
<li><strong><a href="trait.CallBuilder.html">Activities</a></strong>

<ul>
<li>operations to apply to <em>Resources</em></li>
</ul></li>
</ul>

<p>All <em>structures</em> are marked with applicable traits to further categorize them and ease browsing.</p>

<p>Generally speaking, you can invoke <em>Activities</em> like this:</p>
<pre id='rust-example-rendered' class='rust '>
<span class='kw'>let</span> <span class='ident'>r</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>resource</span>().<span class='ident'>activity</span>(...).<span class='ident'>doit</span>()
</pre>

<p>Or specifically ...</p>
<pre id='rust-example-rendered' class='rust '>
<span class='kw'>let</span> <span class='ident'>r</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>annotations</span>().<span class='ident'>search</span>(...).<span class='ident'>doit</span>()
<span class='kw'>let</span> <span class='ident'>r</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>annotations</span>().<span class='ident'>patch</span>(...).<span class='ident'>doit</span>()
<span class='kw'>let</span> <span class='ident'>r</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>annotations</span>().<span class='ident'>create</span>(...).<span class='ident'>doit</span>()
<span class='kw'>let</span> <span class='ident'>r</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>annotations</span>().<span class='ident'>delete</span>(...).<span class='ident'>doit</span>()
<span class='kw'>let</span> <span class='ident'>r</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>annotations</span>().<span class='ident'>update</span>(...).<span class='ident'>doit</span>()
<span class='kw'>let</span> <span class='ident'>r</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>annotations</span>().<span class='ident'>get</span>(...).<span class='ident'>doit</span>()
<span class='kw'>let</span> <span class='ident'>r</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>annotations</span>().<span class='ident'>batch_create</span>(...).<span class='ident'>doit</span>()
</pre>

<p>The <code>resource()</code> and <code>activity(...)</code> calls create <a href="http://en.wikipedia.org/wiki/Builder_pattern">builders</a>. The second one dealing with <code>Activities</code> 
supports various methods to configure the impending operation (not shown here). It is made such that all required arguments have to be 
specified right away (i.e. <code>(...)</code>), whereas all optional ones can be <a href="http://en.wikipedia.org/wiki/Builder_pattern">build up</a> as desired.
The <code>doit()</code> method performs the actual communication with the server and returns the respective result.</p>

<h1 id="usage" class='section-header'><a
                           href="#usage">Usage</a></h1>
<h2 id="setting-up-your-project" class='section-header'><a
                           href="#setting-up-your-project">Setting up your Project</a></h2>
<p>To use this library, you would put the following lines into your <code>Cargo.toml</code> file:</p>

<pre><code class="language-toml">[dependencies]
google-genomics1_beta2 = &quot;*&quot;
</code></pre>

<h2 id="a-complete-example" class='section-header'><a
                           href="#a-complete-example">A complete example</a></h2><pre id='rust-example-rendered' class='rust '>
<span class='kw'>extern</span> <span class='kw'>crate</span> <span class='ident'>hyper</span>;
<span class='kw'>extern</span> <span class='kw'>crate</span> <span class='ident'>yup_oauth2</span> <span class='kw'>as</span> <span class='ident'>oauth2</span>;
<span class='kw'>extern</span> <span class='kw'>crate</span> <span class='ident'>google_genomics1_beta2</span> <span class='kw'>as</span> <span class='ident'>genomics1_beta2</span>;
<span class='kw'>use</span> <span class='ident'>genomics1_beta2</span>::<span class='ident'>Annotation</span>;
<span class='kw'>use</span> <span class='ident'>genomics1_beta2</span>::{<span class='prelude-ty'>Result</span>, <span class='ident'>Error</span>};
<span class='kw'>use</span> <span class='ident'>std</span>::<span class='ident'>default</span>::<span class='ident'>Default</span>;
<span class='kw'>use</span> <span class='ident'>oauth2</span>::{<span class='ident'>Authenticator</span>, <span class='ident'>DefaultAuthenticatorDelegate</span>, <span class='ident'>ApplicationSecret</span>, <span class='ident'>MemoryStorage</span>};
<span class='kw'>use</span> <span class='ident'>genomics1_beta2</span>::<span class='ident'>Genomics</span>;
 
<span class='comment'>// Get an ApplicationSecret instance by some means. It contains the `client_id` and </span>
<span class='comment'>// `client_secret`, among other things.</span>
<span class='kw'>let</span> <span class='ident'>secret</span>: <span class='ident'>ApplicationSecret</span> <span class='op'>=</span> <span class='ident'>Default</span>::<span class='ident'>default</span>();
<span class='comment'>// Instantiate the authenticator. It will choose a suitable authentication flow for you, </span>
<span class='comment'>// unless you replace  `None` with the desired Flow.</span>
<span class='comment'>// Provide your own `AuthenticatorDelegate` to adjust the way it operates and get feedback about </span>
<span class='comment'>// what&#39;s going on. You probably want to bring in your own `TokenStorage` to persist tokens and</span>
<span class='comment'>// retrieve them from storage.</span>
<span class='kw'>let</span> <span class='ident'>auth</span> <span class='op'>=</span> <span class='ident'>Authenticator</span>::<span class='ident'>new</span>(<span class='kw-2'>&amp;</span><span class='ident'>secret</span>, <span class='ident'>DefaultAuthenticatorDelegate</span>,
                              <span class='ident'>hyper</span>::<span class='ident'>Client</span>::<span class='ident'>new</span>(),
                              <span class='op'>&lt;</span><span class='ident'>MemoryStorage</span> <span class='kw'>as</span> <span class='ident'>Default</span><span class='op'>&gt;</span>::<span class='ident'>default</span>(), <span class='prelude-val'>None</span>);
<span class='kw'>let</span> <span class='kw-2'>mut</span> <span class='ident'>hub</span> <span class='op'>=</span> <span class='ident'>Genomics</span>::<span class='ident'>new</span>(<span class='ident'>hyper</span>::<span class='ident'>Client</span>::<span class='ident'>new</span>(), <span class='ident'>auth</span>);
<span class='comment'>// As the method needs a request, you would usually fill it with the desired information</span>
<span class='comment'>// into the respective structure. Some of the parts shown here might not be applicable !</span>
<span class='comment'>// Values shown here are possibly random and not representative !</span>
<span class='kw'>let</span> <span class='kw-2'>mut</span> <span class='ident'>req</span> <span class='op'>=</span> <span class='ident'>Annotation</span>::<span class='ident'>default</span>();
 
<span class='comment'>// You can configure optional parameters by calling the respective setters at will, and</span>
<span class='comment'>// execute the final call using `doit()`.</span>
<span class='comment'>// Values shown here are possibly random and not representative !</span>
<span class='kw'>let</span> <span class='ident'>result</span> <span class='op'>=</span> <span class='ident'>hub</span>.<span class='ident'>annotations</span>().<span class='ident'>patch</span>(<span class='ident'>req</span>, <span class='string'>&quot;annotationId&quot;</span>)
             .<span class='ident'>doit</span>();
 
<span class='kw'>match</span> <span class='ident'>result</span> {
    <span class='prelude-val'>Err</span>(<span class='ident'>e</span>) <span class='op'>=&gt;</span> <span class='kw'>match</span> <span class='ident'>e</span> {
        <span class='comment'>// The Error enum provides details about what exactly happened.</span>
        <span class='comment'>// You can also just use its `Debug`, `Display` or `Error` traits</span>
         <span class='ident'>Error</span>::<span class='ident'>HttpError</span>(_)
        <span class='op'>|</span><span class='ident'>Error</span>::<span class='ident'>MissingAPIKey</span>
        <span class='op'>|</span><span class='ident'>Error</span>::<span class='ident'>MissingToken</span>(_)
        <span class='op'>|</span><span class='ident'>Error</span>::<span class='ident'>Cancelled</span>
        <span class='op'>|</span><span class='ident'>Error</span>::<span class='ident'>UploadSizeLimitExceeded</span>(_, _)
        <span class='op'>|</span><span class='ident'>Error</span>::<span class='ident'>Failure</span>(_)
        <span class='op'>|</span><span class='ident'>Error</span>::<span class='ident'>BadRequest</span>(_)
        <span class='op'>|</span><span class='ident'>Error</span>::<span class='ident'>FieldClash</span>(_)
        <span class='op'>|</span><span class='ident'>Error</span>::<span class='ident'>JsonDecodeError</span>(_, _) <span class='op'>=&gt;</span> <span class='macro'>println</span><span class='macro'>!</span>(<span class='string'>&quot;{}&quot;</span>, <span class='ident'>e</span>),
    },
    <span class='prelude-val'>Ok</span>(<span class='ident'>res</span>) <span class='op'>=&gt;</span> <span class='macro'>println</span><span class='macro'>!</span>(<span class='string'>&quot;Success: {:?}&quot;</span>, <span class='ident'>res</span>),
}
</pre>

<h2 id="handling-errors" class='section-header'><a
                           href="#handling-errors">Handling Errors</a></h2>
<p>All errors produced by the system are provided either as <a href="enum.Result.html">Result</a> enumeration as return value of 
the doit() methods, or handed as possibly intermediate results to either the 
<a href="trait.Delegate.html">Hub Delegate</a>, or the <a href="../yup-oauth2/trait.AuthenticatorDelegate.html">Authenticator Delegate</a>.</p>

<p>When delegates handle errors or intermediate values, they may have a chance to instruct the system to retry. This 
makes the system potentially resilient to all kinds of errors.</p>

<h2 id="uploads-and-downloads" class='section-header'><a
                           href="#uploads-and-downloads">Uploads and Downloads</a></h2>
<p>If a method supports downloads, the response body, which is part of the <a href="enum.Result.html">Result</a>, should be
read by you to obtain the media.
If such a method also supports a <a href="trait.ResponseResult.html">Response Result</a>, it will return that by default.
You can see it as meta-data for the actual media. To trigger a media download, you will have to set up the builder by making
this call: <code>.param(&quot;alt&quot;, &quot;media&quot;)</code>.</p>

<p>Methods supporting uploads can do so using up to 2 different protocols: 
<em>simple</em> and <em>resumable</em>. The distinctiveness of each is represented by customized 
<code>doit(...)</code> methods, which are then named <code>upload(...)</code> and <code>upload_resumable(...)</code> respectively.</p>

<h2 id="customization-and-callbacks" class='section-header'><a
                           href="#customization-and-callbacks">Customization and Callbacks</a></h2>
<p>You may alter the way an <code>doit()</code> method is called by providing a <a href="trait.Delegate.html">delegate</a> to the 
<a href="trait.CallBuilder.html">Method Builder</a> before making the final <code>doit()</code> call. 
Respective methods will be called to provide progress information, as well as determine whether the system should 
retry on failure.</p>

<p>The <a href="trait.Delegate.html">delegate trait</a> is default-implemented, allowing you to customize it with minimal effort.</p>

<h2 id="optional-parts-in-server-requests" class='section-header'><a
                           href="#optional-parts-in-server-requests">Optional Parts in Server-Requests</a></h2>
<p>All structures provided by this library are made to be <a href="trait.RequestValue.html">enocodable</a> and 
<a href="trait.ResponseResult.html">decodable</a> via <em>json</em>. Optionals are used to indicate that partial requests are responses 
are valid.
Most optionals are are considered <a href="trait.Part.html">Parts</a> which are identifiable by name, which will be sent to 
the server to indicate either the set parts of the request or the desired parts in the response.</p>

<h2 id="builder-arguments" class='section-header'><a
                           href="#builder-arguments">Builder Arguments</a></h2>
<p>Using <a href="trait.CallBuilder.html">method builders</a>, you are able to prepare an action call by repeatedly calling it&#39;s methods.
These will always take a single argument, for which the following statements are true.</p>

<ul>
<li><a href="http://en.wikipedia.org/wiki/Plain_old_data_structure">PODs</a> are handed by copy</li>
<li>strings are passed as <code>&amp;str</code></li>
<li><a href="trait.RequestValue.html">request values</a> are moved</li>
</ul>

<p>Arguments will always be copied or cloned into the builder, to make them independent of their original life times.</p>
</div><h2 id='structs' class='section-header'><a href="#structs">Structs</a></h2>
<table>
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AlignReadGroupSetsRequest.html'
                               title='google_genomics1_beta2::AlignReadGroupSetsRequest'>AlignReadGroupSetsRequest</a></td>
                        <td class='docblock short'><p>The read group set align request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AlignReadGroupSetsResponse.html'
                               title='google_genomics1_beta2::AlignReadGroupSetsResponse'>AlignReadGroupSetsResponse</a></td>
                        <td class='docblock short'><p>The read group set align response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Annotation.html'
                               title='google_genomics1_beta2::Annotation'>Annotation</a></td>
                        <td class='docblock short'><p>An annotation describes a region of reference genome. The value of an annotation may be one of several canonical types, supplemented by arbitrary info tags. A variant annotation is represented by one or more of these canonical types. An annotation is not inherently associated with a specific sample or individual (though a client could choose to use annotations in this way). Example canonical annotation types are &#39;Gene&#39; and &#39;Variant&#39;.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationBatchCreateCall.html'
                               title='google_genomics1_beta2::AnnotationBatchCreateCall'>AnnotationBatchCreateCall</a></td>
                        <td class='docblock short'><p>Creates one or more new annotations atomically. All annotations must belong to the same annotation set. Caller must have WRITE permission for this annotation set. For optimal performance, batch positionally adjacent annotations together.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationCreateCall.html'
                               title='google_genomics1_beta2::AnnotationCreateCall'>AnnotationCreateCall</a></td>
                        <td class='docblock short'><p>Creates a new annotation. Caller must have WRITE permission for the associated annotation set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationDeleteCall.html'
                               title='google_genomics1_beta2::AnnotationDeleteCall'>AnnotationDeleteCall</a></td>
                        <td class='docblock short'><p>Deletes an annotation. Caller must have WRITE permission for the associated annotation set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationGetCall.html'
                               title='google_genomics1_beta2::AnnotationGetCall'>AnnotationGetCall</a></td>
                        <td class='docblock short'><p>Gets an annotation. Caller must have READ permission for the associated annotation set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationMethods.html'
                               title='google_genomics1_beta2::AnnotationMethods'>AnnotationMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>annotation</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationPatchCall.html'
                               title='google_genomics1_beta2::AnnotationPatchCall'>AnnotationPatchCall</a></td>
                        <td class='docblock short'><p>Updates an annotation. The update must respect all mutability restrictions and other invariants described on the annotation resource. Caller must have WRITE permission for the associated dataset. This method supports patch semantics.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSearchCall.html'
                               title='google_genomics1_beta2::AnnotationSearchCall'>AnnotationSearchCall</a></td>
                        <td class='docblock short'><p>Searches for annotations that match the given criteria. Results are returned ordered by start position. Annotations that have matching start positions are ordered deterministically. Caller must have READ permission for the queried annotation sets.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSet.html'
                               title='google_genomics1_beta2::AnnotationSet'>AnnotationSet</a></td>
                        <td class='docblock short'><p>An annotation set is a logical grouping of annotations that share consistent type information and provenance. Examples of annotation sets include &#39;all genes from refseq&#39;, and &#39;all variant annotations from ClinVar&#39;.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSetCreateCall.html'
                               title='google_genomics1_beta2::AnnotationSetCreateCall'>AnnotationSetCreateCall</a></td>
                        <td class='docblock short'><p>Creates a new annotation set. Caller must have WRITE permission for the associated dataset.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSetDeleteCall.html'
                               title='google_genomics1_beta2::AnnotationSetDeleteCall'>AnnotationSetDeleteCall</a></td>
                        <td class='docblock short'><p>Deletes an annotation set. Caller must have WRITE permission for the associated annotation set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSetGetCall.html'
                               title='google_genomics1_beta2::AnnotationSetGetCall'>AnnotationSetGetCall</a></td>
                        <td class='docblock short'><p>Gets an annotation set. Caller must have READ permission for the associated dataset.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSetMethods.html'
                               title='google_genomics1_beta2::AnnotationSetMethods'>AnnotationSetMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>annotationSet</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSetPatchCall.html'
                               title='google_genomics1_beta2::AnnotationSetPatchCall'>AnnotationSetPatchCall</a></td>
                        <td class='docblock short'><p>Updates an annotation set. The update must respect all mutability restrictions and other invariants described on the annotation set resource. Caller must have WRITE permission for the associated dataset. This method supports patch semantics.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSetSearchCall.html'
                               title='google_genomics1_beta2::AnnotationSetSearchCall'>AnnotationSetSearchCall</a></td>
                        <td class='docblock short'><p>Searches for annotation sets that match the given criteria. Results are returned in a deterministic order. Caller must have READ permission for the queried datasets.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationSetUpdateCall.html'
                               title='google_genomics1_beta2::AnnotationSetUpdateCall'>AnnotationSetUpdateCall</a></td>
                        <td class='docblock short'><p>Updates an annotation set. The update must respect all mutability restrictions and other invariants described on the annotation set resource. Caller must have WRITE permission for the associated dataset.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.AnnotationUpdateCall.html'
                               title='google_genomics1_beta2::AnnotationUpdateCall'>AnnotationUpdateCall</a></td>
                        <td class='docblock short'><p>Updates an annotation. The update must respect all mutability restrictions and other invariants described on the annotation resource. Caller must have WRITE permission for the associated dataset.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.BatchAnnotationsResponse.html'
                               title='google_genomics1_beta2::BatchAnnotationsResponse'>BatchAnnotationsResponse</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.BatchAnnotationsResponseEntry.html'
                               title='google_genomics1_beta2::BatchAnnotationsResponseEntry'>BatchAnnotationsResponseEntry</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.BatchAnnotationsResponseEntryStatus.html'
                               title='google_genomics1_beta2::BatchAnnotationsResponseEntryStatus'>BatchAnnotationsResponseEntryStatus</a></td>
                        <td class='docblock short'></td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.BatchCreateAnnotationsRequest.html'
                               title='google_genomics1_beta2::BatchCreateAnnotationsRequest'>BatchCreateAnnotationsRequest</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Call.html'
                               title='google_genomics1_beta2::Call'>Call</a></td>
                        <td class='docblock short'><p>A call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a call set with the name NA12345.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallReadGroupSetsRequest.html'
                               title='google_genomics1_beta2::CallReadGroupSetsRequest'>CallReadGroupSetsRequest</a></td>
                        <td class='docblock short'><p>The read group set call request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallReadGroupSetsResponse.html'
                               title='google_genomics1_beta2::CallReadGroupSetsResponse'>CallReadGroupSetsResponse</a></td>
                        <td class='docblock short'><p>The read group set call response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallSet.html'
                               title='google_genomics1_beta2::CallSet'>CallSet</a></td>
                        <td class='docblock short'><p>A call set is a collection of variant calls, typically for one sample. It belongs to a variant set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallsetCreateCall.html'
                               title='google_genomics1_beta2::CallsetCreateCall'>CallsetCreateCall</a></td>
                        <td class='docblock short'><p>Creates a new call set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallsetDeleteCall.html'
                               title='google_genomics1_beta2::CallsetDeleteCall'>CallsetDeleteCall</a></td>
                        <td class='docblock short'><p>Deletes a call set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallsetGetCall.html'
                               title='google_genomics1_beta2::CallsetGetCall'>CallsetGetCall</a></td>
                        <td class='docblock short'><p>Gets a call set by ID.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallsetMethods.html'
                               title='google_genomics1_beta2::CallsetMethods'>CallsetMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>callset</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallsetPatchCall.html'
                               title='google_genomics1_beta2::CallsetPatchCall'>CallsetPatchCall</a></td>
                        <td class='docblock short'><p>Updates a call set. This method supports patch semantics.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallsetSearchCall.html'
                               title='google_genomics1_beta2::CallsetSearchCall'>CallsetSearchCall</a></td>
                        <td class='docblock short'><p>Gets a list of call sets matching the criteria.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CallsetUpdateCall.html'
                               title='google_genomics1_beta2::CallsetUpdateCall'>CallsetUpdateCall</a></td>
                        <td class='docblock short'><p>Updates a call set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CigarUnit.html'
                               title='google_genomics1_beta2::CigarUnit'>CigarUnit</a></td>
                        <td class='docblock short'><p>A single CIGAR operation.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.CoverageBucket.html'
                               title='google_genomics1_beta2::CoverageBucket'>CoverageBucket</a></td>
                        <td class='docblock short'><p>A bucket over which read coverage has been precomputed. A bucket corresponds to a specific range of the reference sequence.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Dataset.html'
                               title='google_genomics1_beta2::Dataset'>Dataset</a></td>
                        <td class='docblock short'><p>A Dataset is a collection of genomic data.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DatasetCreateCall.html'
                               title='google_genomics1_beta2::DatasetCreateCall'>DatasetCreateCall</a></td>
                        <td class='docblock short'><p>Creates a new dataset.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DatasetDeleteCall.html'
                               title='google_genomics1_beta2::DatasetDeleteCall'>DatasetDeleteCall</a></td>
                        <td class='docblock short'><p>Deletes a dataset.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DatasetGetCall.html'
                               title='google_genomics1_beta2::DatasetGetCall'>DatasetGetCall</a></td>
                        <td class='docblock short'><p>Gets a dataset by ID.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DatasetListCall.html'
                               title='google_genomics1_beta2::DatasetListCall'>DatasetListCall</a></td>
                        <td class='docblock short'><p>Lists datasets within a project.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DatasetMethods.html'
                               title='google_genomics1_beta2::DatasetMethods'>DatasetMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>dataset</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DatasetPatchCall.html'
                               title='google_genomics1_beta2::DatasetPatchCall'>DatasetPatchCall</a></td>
                        <td class='docblock short'><p>Updates a dataset. This method supports patch semantics.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DatasetUndeleteCall.html'
                               title='google_genomics1_beta2::DatasetUndeleteCall'>DatasetUndeleteCall</a></td>
                        <td class='docblock short'><p>Undeletes a dataset by restoring a dataset which was deleted via this API. This operation is only possible for a week after the deletion occurred.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DatasetUpdateCall.html'
                               title='google_genomics1_beta2::DatasetUpdateCall'>DatasetUpdateCall</a></td>
                        <td class='docblock short'><p>Updates a dataset.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.DefaultDelegate.html'
                               title='google_genomics1_beta2::DefaultDelegate'>DefaultDelegate</a></td>
                        <td class='docblock short'><p>A delegate with a conservative default implementation, which is used if no other delegate is
set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ErrorResponse.html'
                               title='google_genomics1_beta2::ErrorResponse'>ErrorResponse</a></td>
                        <td class='docblock short'><p>A utility to represent detailed errors we might see in case there are BadRequests.
The latter happen if the sent parameters or request structures are unsound</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExperimentalCreateJobRequest.html'
                               title='google_genomics1_beta2::ExperimentalCreateJobRequest'>ExperimentalCreateJobRequest</a></td>
                        <td class='docblock short'><p>The job creation request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExperimentalCreateJobResponse.html'
                               title='google_genomics1_beta2::ExperimentalCreateJobResponse'>ExperimentalCreateJobResponse</a></td>
                        <td class='docblock short'><p>The job creation response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExperimentalJobCreateCall.html'
                               title='google_genomics1_beta2::ExperimentalJobCreateCall'>ExperimentalJobCreateCall</a></td>
                        <td class='docblock short'><p>Creates and asynchronously runs an ad-hoc job. This is an experimental call and may be removed or changed at any time.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExperimentalMethods.html'
                               title='google_genomics1_beta2::ExperimentalMethods'>ExperimentalMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>experimental</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExportReadGroupSetsRequest.html'
                               title='google_genomics1_beta2::ExportReadGroupSetsRequest'>ExportReadGroupSetsRequest</a></td>
                        <td class='docblock short'><p>The read group set export request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExportReadGroupSetsResponse.html'
                               title='google_genomics1_beta2::ExportReadGroupSetsResponse'>ExportReadGroupSetsResponse</a></td>
                        <td class='docblock short'><p>The read group set export response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExportVariantSetRequest.html'
                               title='google_genomics1_beta2::ExportVariantSetRequest'>ExportVariantSetRequest</a></td>
                        <td class='docblock short'><p>The variant data export request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExportVariantSetResponse.html'
                               title='google_genomics1_beta2::ExportVariantSetResponse'>ExportVariantSetResponse</a></td>
                        <td class='docblock short'><p>The variant data export response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ExternalId.html'
                               title='google_genomics1_beta2::ExternalId'>ExternalId</a></td>
                        <td class='docblock short'></td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.FastqMetadata.html'
                               title='google_genomics1_beta2::FastqMetadata'>FastqMetadata</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Genomics.html'
                               title='google_genomics1_beta2::Genomics'>Genomics</a></td>
                        <td class='docblock short'><p>Central instance to access all Genomics related resource activities</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ImportReadGroupSetsRequest.html'
                               title='google_genomics1_beta2::ImportReadGroupSetsRequest'>ImportReadGroupSetsRequest</a></td>
                        <td class='docblock short'><p>The read group set import request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ImportReadGroupSetsResponse.html'
                               title='google_genomics1_beta2::ImportReadGroupSetsResponse'>ImportReadGroupSetsResponse</a></td>
                        <td class='docblock short'><p>The read group set import response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ImportVariantsRequest.html'
                               title='google_genomics1_beta2::ImportVariantsRequest'>ImportVariantsRequest</a></td>
                        <td class='docblock short'><p>The variant data import request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ImportVariantsResponse.html'
                               title='google_genomics1_beta2::ImportVariantsResponse'>ImportVariantsResponse</a></td>
                        <td class='docblock short'><p>The variant data import response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Int32Value.html'
                               title='google_genomics1_beta2::Int32Value'>Int32Value</a></td>
                        <td class='docblock short'><p>Wrapper message for int32.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.InterleavedFastqSource.html'
                               title='google_genomics1_beta2::InterleavedFastqSource'>InterleavedFastqSource</a></td>
                        <td class='docblock short'><p>Describes an interleaved FASTQ file source for alignment.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Job.html'
                               title='google_genomics1_beta2::Job'>Job</a></td>
                        <td class='docblock short'><p>A Job represents an ongoing process that can be monitored for status information.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.JobCancelCall.html'
                               title='google_genomics1_beta2::JobCancelCall'>JobCancelCall</a></td>
                        <td class='docblock short'><p>Cancels a job by ID. Note that it is possible for partial results to be generated and stored for cancelled jobs.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.JobGetCall.html'
                               title='google_genomics1_beta2::JobGetCall'>JobGetCall</a></td>
                        <td class='docblock short'><p>Gets a job by ID.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.JobMethods.html'
                               title='google_genomics1_beta2::JobMethods'>JobMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>job</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.JobRequest.html'
                               title='google_genomics1_beta2::JobRequest'>JobRequest</a></td>
                        <td class='docblock short'><p>A summary representation of the service request that spawned the job.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.JobSearchCall.html'
                               title='google_genomics1_beta2::JobSearchCall'>JobSearchCall</a></td>
                        <td class='docblock short'><p>Gets a list of jobs matching the criteria.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.LinearAlignment.html'
                               title='google_genomics1_beta2::LinearAlignment'>LinearAlignment</a></td>
                        <td class='docblock short'><p>A linear alignment can be represented by one CIGAR string. Describes the mapped position and local alignment of the read to the reference.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ListBasesResponse.html'
                               title='google_genomics1_beta2::ListBasesResponse'>ListBasesResponse</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ListCoverageBucketsResponse.html'
                               title='google_genomics1_beta2::ListCoverageBucketsResponse'>ListCoverageBucketsResponse</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ListDatasetsResponse.html'
                               title='google_genomics1_beta2::ListDatasetsResponse'>ListDatasetsResponse</a></td>
                        <td class='docblock short'><p>The dataset list response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.MergeVariantsRequest.html'
                               title='google_genomics1_beta2::MergeVariantsRequest'>MergeVariantsRequest</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Metadata.html'
                               title='google_genomics1_beta2::Metadata'>Metadata</a></td>
                        <td class='docblock short'><p>Metadata describes a single piece of variant call metadata. These data include a top level key and either a single value string (value) or a list of key-value pairs (info.) Value and info are mutually exclusive.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.MethodInfo.html'
                               title='google_genomics1_beta2::MethodInfo'>MethodInfo</a></td>
                        <td class='docblock short'><p>Contains information about an API request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.MultiPartReader.html'
                               title='google_genomics1_beta2::MultiPartReader'>MultiPartReader</a></td>
                        <td class='docblock short'><p>Provides a <code>Read</code> interface that converts multiple parts into the protocol
identified by <a href="https://tools.ietf.org/html/rfc2387">RFC2387</a>.
<strong>Note</strong>: This implementation is just as rich as it needs to be to perform uploads
to google APIs, and might not be a fully-featured implementation.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.PairedFastqSource.html'
                               title='google_genomics1_beta2::PairedFastqSource'>PairedFastqSource</a></td>
                        <td class='docblock short'><p>Describes a paired-end FASTQ file source for alignment.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Position.html'
                               title='google_genomics1_beta2::Position'>Position</a></td>
                        <td class='docblock short'><p>An abstraction for referring to a genomic position, in relation to some already known reference. For now, represents a genomic position as a reference name, a base number on that reference (0-based), and a determination of forward or reverse strand.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.QueryRange.html'
                               title='google_genomics1_beta2::QueryRange'>QueryRange</a></td>
                        <td class='docblock short'><p>A 0-based half-open genomic coordinate range for search requests.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Range.html'
                               title='google_genomics1_beta2::Range'>Range</a></td>
                        <td class='docblock short'><p>A 0-based half-open genomic coordinate range over a reference sequence.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.RangePosition.html'
                               title='google_genomics1_beta2::RangePosition'>RangePosition</a></td>
                        <td class='docblock short'><p>A 0-based half-open genomic coordinate range over a reference sequence, for representing the position of a genomic resource.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Read.html'
                               title='google_genomics1_beta2::Read'>Read</a></td>
                        <td class='docblock short'><p>A read alignment describes a linear alignment of a string of DNA to a reference sequence, in addition to metadata about the fragment (the molecule of DNA sequenced) and the read (the bases which were read by the sequencer). A read is equivalent to a line in a SAM file. A read belongs to exactly one read group and exactly one read group set. Generating a reference-aligned sequence string When interacting with mapped reads, it&#39;s often useful to produce a string representing the local alignment of the read to reference. The following pseudocode demonstrates one way of doing this:
out = &quot;&quot; offset = 0 for c in read.alignment.cigar { switch c.operation { case &quot;ALIGNMENT_MATCH&quot;, &quot;SEQUENCE_MATCH&quot;, &quot;SEQUENCE_MISMATCH&quot;: out += read.alignedSequence[offset:offset+c.operationLength] offset += c.operationLength break case &quot;CLIP_SOFT&quot;, &quot;INSERT&quot;: offset += c.operationLength break case &quot;PAD&quot;: out += repeat(&quot;*&quot;, c.operationLength) break case &quot;DELETE&quot;: out += repeat(&quot;-&quot;, c.operationLength) break case &quot;SKIP&quot;: out += repeat(&quot; &quot;, c.operationLength) break case &quot;CLIP_HARD&quot;: break } } return out
Converting to SAM&#39;s CIGAR string The following pseudocode generates a SAM CIGAR string from the cigar field. Note that this is a lossy conversion (cigar.referenceSequence is lost).
cigarMap = { &quot;ALIGNMENT_MATCH&quot;: &quot;M&quot;, &quot;INSERT&quot;: &quot;I&quot;, &quot;DELETE&quot;: &quot;D&quot;, &quot;SKIP&quot;: &quot;N&quot;, &quot;CLIP_SOFT&quot;: &quot;S&quot;, &quot;CLIP_HARD&quot;: &quot;H&quot;, &quot;PAD&quot;: &quot;P&quot;, &quot;SEQUENCE_MATCH&quot;: &quot;=&quot;, &quot;SEQUENCE_MISMATCH&quot;: &quot;X&quot;, } cigarStr = &quot;&quot; for c in read.alignment.cigar { cigarStr += c.operationLength + cigarMap[c.operation] } return cigarStr</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadGroup.html'
                               title='google_genomics1_beta2::ReadGroup'>ReadGroup</a></td>
                        <td class='docblock short'><p>A read group is all the data that&#39;s processed the same way by the sequencer.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadGroupExperiment.html'
                               title='google_genomics1_beta2::ReadGroupExperiment'>ReadGroupExperiment</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadGroupProgram.html'
                               title='google_genomics1_beta2::ReadGroupProgram'>ReadGroupProgram</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadGroupSet.html'
                               title='google_genomics1_beta2::ReadGroupSet'>ReadGroupSet</a></td>
                        <td class='docblock short'><p>A read group set is a logical collection of read groups, which are collections of reads produced by a sequencer. A read group set typically models reads corresponding to one sample, sequenced one way, and aligned one way.
- A read group set belongs to one dataset.
- A read group belongs to one read group set.
- A read belongs to one read group.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadMethods.html'
                               title='google_genomics1_beta2::ReadMethods'>ReadMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>read</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadSearchCall.html'
                               title='google_genomics1_beta2::ReadSearchCall'>ReadSearchCall</a></td>
                        <td class='docblock short'><p>Gets a list of reads for one or more read group sets. Reads search operates over a genomic coordinate space of reference sequence &amp; position defined over the reference sequences to which the requested read group sets are aligned.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetAlignCall.html'
                               title='google_genomics1_beta2::ReadgroupsetAlignCall'>ReadgroupsetAlignCall</a></td>
                        <td class='docblock short'><p>Aligns read data from existing read group sets or files from Google Cloud Storage. See the  alignment and variant calling documentation for more details.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetCallCall.html'
                               title='google_genomics1_beta2::ReadgroupsetCallCall'>ReadgroupsetCallCall</a></td>
                        <td class='docblock short'><p>Calls variants on read data from existing read group sets or files from Google Cloud Storage. See the  alignment and variant calling documentation for more details.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetCoveragebucketListCall.html'
                               title='google_genomics1_beta2::ReadgroupsetCoveragebucketListCall'>ReadgroupsetCoveragebucketListCall</a></td>
                        <td class='docblock short'><p>Lists fixed width coverage buckets for a read group set, each of which correspond to a range of a reference sequence. Each bucket summarizes coverage information across its corresponding genomic range.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetDeleteCall.html'
                               title='google_genomics1_beta2::ReadgroupsetDeleteCall'>ReadgroupsetDeleteCall</a></td>
                        <td class='docblock short'><p>Deletes a read group set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetExportCall.html'
                               title='google_genomics1_beta2::ReadgroupsetExportCall'>ReadgroupsetExportCall</a></td>
                        <td class='docblock short'><p>Exports read group sets to a BAM file in Google Cloud Storage.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetGetCall.html'
                               title='google_genomics1_beta2::ReadgroupsetGetCall'>ReadgroupsetGetCall</a></td>
                        <td class='docblock short'><p>Gets a read group set by ID.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetImportCall.html'
                               title='google_genomics1_beta2::ReadgroupsetImportCall'>ReadgroupsetImportCall</a></td>
                        <td class='docblock short'><p>Creates read group sets by asynchronously importing the provided information.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetMethods.html'
                               title='google_genomics1_beta2::ReadgroupsetMethods'>ReadgroupsetMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>readgroupset</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetPatchCall.html'
                               title='google_genomics1_beta2::ReadgroupsetPatchCall'>ReadgroupsetPatchCall</a></td>
                        <td class='docblock short'><p>Updates a read group set. This method supports patch semantics.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetSearchCall.html'
                               title='google_genomics1_beta2::ReadgroupsetSearchCall'>ReadgroupsetSearchCall</a></td>
                        <td class='docblock short'><p>Searches for read group sets matching the criteria.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReadgroupsetUpdateCall.html'
                               title='google_genomics1_beta2::ReadgroupsetUpdateCall'>ReadgroupsetUpdateCall</a></td>
                        <td class='docblock short'><p>Updates a read group set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Reference.html'
                               title='google_genomics1_beta2::Reference'>Reference</a></td>
                        <td class='docblock short'><p>A reference is a canonical assembled DNA sequence, intended to act as a reference coordinate space for other genomic annotations. A single reference might represent the human chromosome 1 or mitochandrial DNA, for instance. A reference belongs to one or more reference sets.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferenceBaseListCall.html'
                               title='google_genomics1_beta2::ReferenceBaseListCall'>ReferenceBaseListCall</a></td>
                        <td class='docblock short'><p>Lists the bases in a reference, optionally restricted to a range.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferenceBound.html'
                               title='google_genomics1_beta2::ReferenceBound'>ReferenceBound</a></td>
                        <td class='docblock short'><p>ReferenceBound records an upper bound for the starting coordinate of variants in a particular reference.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferenceGetCall.html'
                               title='google_genomics1_beta2::ReferenceGetCall'>ReferenceGetCall</a></td>
                        <td class='docblock short'><p>Gets a reference.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferenceMethods.html'
                               title='google_genomics1_beta2::ReferenceMethods'>ReferenceMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>reference</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferenceSearchCall.html'
                               title='google_genomics1_beta2::ReferenceSearchCall'>ReferenceSearchCall</a></td>
                        <td class='docblock short'><p>Searches for references which match the given criteria.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferenceSet.html'
                               title='google_genomics1_beta2::ReferenceSet'>ReferenceSet</a></td>
                        <td class='docblock short'><p>A reference set is a set of references which typically comprise a reference assembly for a species, such as GRCh38 which is representative of the human genome. A reference set defines a common coordinate space for comparing reference-aligned experimental data. A reference set contains 1 or more references.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferencesetGetCall.html'
                               title='google_genomics1_beta2::ReferencesetGetCall'>ReferencesetGetCall</a></td>
                        <td class='docblock short'><p>Gets a reference set.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferencesetMethods.html'
                               title='google_genomics1_beta2::ReferencesetMethods'>ReferencesetMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>referenceset</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.ReferencesetSearchCall.html'
                               title='google_genomics1_beta2::ReferencesetSearchCall'>ReferencesetSearchCall</a></td>
                        <td class='docblock short'><p>Searches for reference sets which match the given criteria.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchAnnotationSetsRequest.html'
                               title='google_genomics1_beta2::SearchAnnotationSetsRequest'>SearchAnnotationSetsRequest</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchAnnotationSetsResponse.html'
                               title='google_genomics1_beta2::SearchAnnotationSetsResponse'>SearchAnnotationSetsResponse</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchAnnotationsRequest.html'
                               title='google_genomics1_beta2::SearchAnnotationsRequest'>SearchAnnotationsRequest</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchAnnotationsResponse.html'
                               title='google_genomics1_beta2::SearchAnnotationsResponse'>SearchAnnotationsResponse</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchCallSetsRequest.html'
                               title='google_genomics1_beta2::SearchCallSetsRequest'>SearchCallSetsRequest</a></td>
                        <td class='docblock short'><p>The call set search request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchCallSetsResponse.html'
                               title='google_genomics1_beta2::SearchCallSetsResponse'>SearchCallSetsResponse</a></td>
                        <td class='docblock short'><p>The call set search response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchJobsRequest.html'
                               title='google_genomics1_beta2::SearchJobsRequest'>SearchJobsRequest</a></td>
                        <td class='docblock short'><p>The jobs search request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchJobsResponse.html'
                               title='google_genomics1_beta2::SearchJobsResponse'>SearchJobsResponse</a></td>
                        <td class='docblock short'><p>The job search response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchReadGroupSetsRequest.html'
                               title='google_genomics1_beta2::SearchReadGroupSetsRequest'>SearchReadGroupSetsRequest</a></td>
                        <td class='docblock short'><p>The read group set search request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchReadGroupSetsResponse.html'
                               title='google_genomics1_beta2::SearchReadGroupSetsResponse'>SearchReadGroupSetsResponse</a></td>
                        <td class='docblock short'><p>The read group set search response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchReadsRequest.html'
                               title='google_genomics1_beta2::SearchReadsRequest'>SearchReadsRequest</a></td>
                        <td class='docblock short'><p>The read search request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchReadsResponse.html'
                               title='google_genomics1_beta2::SearchReadsResponse'>SearchReadsResponse</a></td>
                        <td class='docblock short'><p>The read search response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchReferenceSetsRequest.html'
                               title='google_genomics1_beta2::SearchReferenceSetsRequest'>SearchReferenceSetsRequest</a></td>
                        <td class='docblock short'></td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchReferenceSetsResponse.html'
                               title='google_genomics1_beta2::SearchReferenceSetsResponse'>SearchReferenceSetsResponse</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchReferencesRequest.html'
                               title='google_genomics1_beta2::SearchReferencesRequest'>SearchReferencesRequest</a></td>
                        <td class='docblock short'></td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchReferencesResponse.html'
                               title='google_genomics1_beta2::SearchReferencesResponse'>SearchReferencesResponse</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchVariantSetsRequest.html'
                               title='google_genomics1_beta2::SearchVariantSetsRequest'>SearchVariantSetsRequest</a></td>
                        <td class='docblock short'><p>The search variant sets request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchVariantSetsResponse.html'
                               title='google_genomics1_beta2::SearchVariantSetsResponse'>SearchVariantSetsResponse</a></td>
                        <td class='docblock short'><p>The search variant sets response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchVariantsRequest.html'
                               title='google_genomics1_beta2::SearchVariantsRequest'>SearchVariantsRequest</a></td>
                        <td class='docblock short'><p>The variant search request.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.SearchVariantsResponse.html'
                               title='google_genomics1_beta2::SearchVariantsResponse'>SearchVariantsResponse</a></td>
                        <td class='docblock short'><p>The variant search response.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.StreamReadsRequest.html'
                               title='google_genomics1_beta2::StreamReadsRequest'>StreamReadsRequest</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.StreamReadsResponse.html'
                               title='google_genomics1_beta2::StreamReadsResponse'>StreamReadsResponse</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.StreamingReadstoreMethods.html'
                               title='google_genomics1_beta2::StreamingReadstoreMethods'>StreamingReadstoreMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>streamingReadstore</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.StreamingReadstoreStreamreadCall.html'
                               title='google_genomics1_beta2::StreamingReadstoreStreamreadCall'>StreamingReadstoreStreamreadCall</a></td>
                        <td class='docblock short'><p>Gets a stream of reads for one or more read group sets. Reads search operates over a genomic coordinate space of reference sequence &amp; position defined over the reference sequences to which the requested read group sets are aligned.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Transcript.html'
                               title='google_genomics1_beta2::Transcript'>Transcript</a></td>
                        <td class='docblock short'><p>A transcript represents the assertion that a particular region of the reference genome may be transcribed as RNA.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.TranscriptCodingSequence.html'
                               title='google_genomics1_beta2::TranscriptCodingSequence'>TranscriptCodingSequence</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.TranscriptExon.html'
                               title='google_genomics1_beta2::TranscriptExon'>TranscriptExon</a></td>
                        <td class='docblock short'><p>There is no detailed description.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.Variant.html'
                               title='google_genomics1_beta2::Variant'>Variant</a></td>
                        <td class='docblock short'><p>A variant represents a change in DNA sequence relative to a reference sequence. For example, a variant could represent a SNP or an insertion. Variants belong to a variant set. Each of the calls on a variant represent a determination of genotype with respect to that variant. For example, a call might assign probability of 0.32 to the occurrence of a SNP named rs1234 in a sample named NA12345. A call belongs to a call set, which contains related calls typically from one sample.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantAnnotation.html'
                               title='google_genomics1_beta2::VariantAnnotation'>VariantAnnotation</a></td>
                        <td class='docblock short'><p>A Variant annotation.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantAnnotationCondition.html'
                               title='google_genomics1_beta2::VariantAnnotationCondition'>VariantAnnotationCondition</a></td>
                        <td class='docblock short'></td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantCreateCall.html'
                               title='google_genomics1_beta2::VariantCreateCall'>VariantCreateCall</a></td>
                        <td class='docblock short'><p>Creates a new variant.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantDeleteCall.html'
                               title='google_genomics1_beta2::VariantDeleteCall'>VariantDeleteCall</a></td>
                        <td class='docblock short'><p>Deletes a variant.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantGetCall.html'
                               title='google_genomics1_beta2::VariantGetCall'>VariantGetCall</a></td>
                        <td class='docblock short'><p>Gets a variant by ID.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantMethods.html'
                               title='google_genomics1_beta2::VariantMethods'>VariantMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>variant</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantSearchCall.html'
                               title='google_genomics1_beta2::VariantSearchCall'>VariantSearchCall</a></td>
                        <td class='docblock short'><p>Gets a list of variants matching the criteria.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantSet.html'
                               title='google_genomics1_beta2::VariantSet'>VariantSet</a></td>
                        <td class='docblock short'><p>A variant set is a collection of call sets and variants. It contains summary statistics of those contents. A variant set belongs to a dataset.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantUpdateCall.html'
                               title='google_genomics1_beta2::VariantUpdateCall'>VariantUpdateCall</a></td>
                        <td class='docblock short'><p>Updates a variant&#39;s names and info fields. All other modifications are silently ignored. Returns the modified variant without its calls.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetDeleteCall.html'
                               title='google_genomics1_beta2::VariantsetDeleteCall'>VariantsetDeleteCall</a></td>
                        <td class='docblock short'><p>Deletes the contents of a variant set. The variant set object is not deleted.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetExportCall.html'
                               title='google_genomics1_beta2::VariantsetExportCall'>VariantsetExportCall</a></td>
                        <td class='docblock short'><p>Exports variant set data to an external destination.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetGetCall.html'
                               title='google_genomics1_beta2::VariantsetGetCall'>VariantsetGetCall</a></td>
                        <td class='docblock short'><p>Gets a variant set by ID.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetImportVariantCall.html'
                               title='google_genomics1_beta2::VariantsetImportVariantCall'>VariantsetImportVariantCall</a></td>
                        <td class='docblock short'><p>Creates variant data by asynchronously importing the provided information.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetMergeVariantCall.html'
                               title='google_genomics1_beta2::VariantsetMergeVariantCall'>VariantsetMergeVariantCall</a></td>
                        <td class='docblock short'><p>Merges the given variants with existing variants. Each variant will be merged with an existing variant that matches its reference sequence, start, end, reference bases, and alternative bases. If no such variant exists, a new one will be created.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetMethods.html'
                               title='google_genomics1_beta2::VariantsetMethods'>VariantsetMethods</a></td>
                        <td class='docblock short'><p>A builder providing access to all methods supported on <em>variantset</em> resources.
It is not used directly, but through the <code>Genomics</code> hub.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetPatchCall.html'
                               title='google_genomics1_beta2::VariantsetPatchCall'>VariantsetPatchCall</a></td>
                        <td class='docblock short'><p>Updates a variant set&#39;s metadata. All other modifications are silently ignored. This method supports patch semantics.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetSearchCall.html'
                               title='google_genomics1_beta2::VariantsetSearchCall'>VariantsetSearchCall</a></td>
                        <td class='docblock short'><p>Returns a list of all variant sets matching search criteria.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='struct' href='struct.VariantsetUpdateCall.html'
                               title='google_genomics1_beta2::VariantsetUpdateCall'>VariantsetUpdateCall</a></td>
                        <td class='docblock short'><p>Updates a variant set&#39;s metadata. All other modifications are silently ignored.</p>
</td>
                    </tr>
                </table><h2 id='enums' class='section-header'><a href="#enums">Enums</a></h2>
<table>
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='enum' href='enum.Error.html'
                               title='google_genomics1_beta2::Error'>Error</a></td>
                        <td class='docblock short'></td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='enum' href='enum.Scope.html'
                               title='google_genomics1_beta2::Scope'>Scope</a></td>
                        <td class='docblock short'><p>Identifies the an OAuth2 authorization scope.
A scope is needed when requesting an
<a href="https://developers.google.com/youtube/v3/guides/authentication">authorization token</a>.</p>
</td>
                    </tr>
                </table><h2 id='traits' class='section-header'><a href="#traits">Traits</a></h2>
<table>
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.CallBuilder.html'
                               title='google_genomics1_beta2::CallBuilder'>CallBuilder</a></td>
                        <td class='docblock short'><p>Identifies types which represent builders for a particular resource method</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.Delegate.html'
                               title='google_genomics1_beta2::Delegate'>Delegate</a></td>
                        <td class='docblock short'><p>A trait specifying functionality to help controlling any request performed by the API.
The trait has a conservative default implementation.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.Hub.html'
                               title='google_genomics1_beta2::Hub'>Hub</a></td>
                        <td class='docblock short'><p>Identifies the Hub. There is only one per library, this trait is supposed
to make intended use more explicit.
The hub allows to access all resource methods more easily.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.MethodsBuilder.html'
                               title='google_genomics1_beta2::MethodsBuilder'>MethodsBuilder</a></td>
                        <td class='docblock short'><p>Identifies types for building methods of a particular resource type</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.NestedType.html'
                               title='google_genomics1_beta2::NestedType'>NestedType</a></td>
                        <td class='docblock short'><p>Identifies types which are only used by other types internally.
They have no special meaning, this trait just marks them for completeness.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.Part.html'
                               title='google_genomics1_beta2::Part'>Part</a></td>
                        <td class='docblock short'><p>Identifies types which are only used as part of other types, which
usually are carrying the <code>Resource</code> trait.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.ReadSeek.html'
                               title='google_genomics1_beta2::ReadSeek'>ReadSeek</a></td>
                        <td class='docblock short'><p>A utility to specify reader types which provide seeking capabilities too</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.RequestValue.html'
                               title='google_genomics1_beta2::RequestValue'>RequestValue</a></td>
                        <td class='docblock short'><p>Identifies types which are used in API requests.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.Resource.html'
                               title='google_genomics1_beta2::Resource'>Resource</a></td>
                        <td class='docblock short'><p>Identifies types which can be inserted and deleted.
Types with this trait are most commonly used by clients of this API.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.ResponseResult.html'
                               title='google_genomics1_beta2::ResponseResult'>ResponseResult</a></td>
                        <td class='docblock short'><p>Identifies types which are used in API responses.</p>
</td>
                    </tr>
                
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='trait' href='trait.ToParts.html'
                               title='google_genomics1_beta2::ToParts'>ToParts</a></td>
                        <td class='docblock short'><p>A trait for all types that can convert themselves into a <em>parts</em> string</p>
</td>
                    </tr>
                </table><h2 id='functions' class='section-header'><a href="#functions">Functions</a></h2>
<table>
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='fn' href='fn.remove_json_null_values.html'
                               title='google_genomics1_beta2::remove_json_null_values'>remove_json_null_values</a></td>
                        <td class='docblock short'></td>
                    </tr>
                </table><h2 id='types' class='section-header'><a href="#types">Type Definitions</a></h2>
<table>
                    <tr>
                        <td><a class='stability Unmarked' title='No stability level'></a><a class='type' href='type.Result.html'
                               title='google_genomics1_beta2::Result'>Result</a></td>
                        <td class='docblock short'><p>A universal result type used as return for all calls.</p>
</td>
                    </tr>
                </table></section>
    <section id='search' class="content hidden"></section>

    <section class="footer"></section>

    <div id="help" class="hidden">
        <div class="shortcuts">
            <h1>Keyboard shortcuts</h1>
            <dl>
                <dt>?</dt>
                <dd>Show this help dialog</dd>
                <dt>S</dt>
                <dd>Focus the search field</dd>
                <dt>&larrb;</dt>
                <dd>Move up in search results</dd>
                <dt>&rarrb;</dt>
                <dd>Move down in search results</dd>
                <dt>&#9166;</dt>
                <dd>Go to active search result</dd>
            </dl>
        </div>
        <div class="infos">
            <h1>Search tricks</h1>
            <p>
                Prefix searches with a type followed by a colon (e.g.
                <code>fn:</code>) to restrict the search to a given type.
            </p>
            <p>
                Accepted types are: <code>fn</code>, <code>mod</code>,
                <code>struct</code>, <code>enum</code>,
                <code>trait</code>, <code>typedef</code> (or
                <code>tdef</code>).
            </p>
        </div>
    </div>

    

    <script>
        window.rootPath = "../";
        window.currentCrate = "google_genomics1_beta2";
        window.playgroundUrl = "";
    </script>
    <script src="../jquery.js"></script>
    <script src="../main.js"></script>
    
    <script async src="../search-index.js"></script>
</body>
</html>